### PR TITLE
api: Add Unit Serial Number page (80h) support

### PR DESCRIPTION
**Pull request by lxbsz**
Signed-off-by: Xiubo Li <lixiubo@cmss.chinamobile.com>

### CODE REVIEW

**Comment by lxbsz** [comment]
Test：
[root@localhost ~]# sg_vpd /dev/sdc --page=0x80
Unit serial number VPD page:
  Unit serial number: 8c913dc4-a136-4413-a926-3703df4c15b9
[root@localhost ~]#

File: api.c [hunk 1]
```diff
@@ -348,6 +349,32 @@ int tcmu_emulate_evpd_inquiry(
...
+		data[1] = 0x80;
+
+		wwn = tcmu_get_wwn(dev);
+		if (!wwn) {
+			return tcmu_set_sense_data(sense, HARDWARE_ERROR,
+						   ASC_INTERNAL_TARGET_FAILURE,
+						   NULL);
+		}
+
+		len = sprintf(&data[4], "%s", wwn);
```
Comment (seroyer) [inline api.c]: How about snprintf here?  Just ensures that in case the wwn ever mysteriously returns > 508 bytes, you won't corrupt the stack.

Comment (lxbsz) [inline api.c]: Actually in kernel, the unit_serial has been limited to INQUIRY_VPD_SERIAL_LEN, it is 254. So if every thing works correctly, this shouldn't happen.

But as it maybe mysteriouly corrupt the data for some unknown reason(for now it won't happen for the current code), so I will limit it to 256 using snprintf().

Thanks,

**Comment by lxbsz** [comment]
@seroyer 
Has been updated， please review.
Thanks,

**Comment by mikechristie** [comment]
Patch looks ok. Will give seroyer a day to comment.

**Comment by seroyer** [comment]
Looks good to me

**Comment by mikechristie** [comment]
Thanks.

**Pull request merged**